HACKING.md

```diff
@@ -158,12 +158,12 @@ To test the `snapd` REST API daemon on a snappy system you need to
 transfer it to the snappy system and then run:
 
     sudo systemctl stop snapd.service snapd.socket
-    sudo /lib/systemd/systemd-activate -E SNAPD_DEBUG=3 -E SNAP_REEXEC=0 -E SNAPD_DEBUG_HTTP=3 -l /run/snapd.socket -l /run/snapd-snap.socket ./snapd
+    sudo /lib/systemd/systemd-activate -E SNAPD_DEBUG=3 -E SNAPD_DEBUG_HTTP=3 -l /run/snapd.socket -l /run/snapd-snap.socket ./snapd
```

Comment:
Yes, I should have explained this better. I think we do not need it for two reasons:
- cmd.go checks for /proc/self/exe and tries to find that path under the core snaps base dir - if running from a git checkout there will be no /snap/core/current/home/user/go/src/github.com... so it won't re-exec
- the code checks if the re-exec snapd is older and not re-exec if it is older

However I am happy to revert this change if it feels safer to have it.